crates/astria-gossipnet/src/network.rs

```diff
@@ -85,33 +92,31 @@ impl NetworkBuilder {
         self
     }
 
-    pub fn build(self) -> Result<Network> {
-        Network::new(self.bootnodes, self.port)
+    pub fn keypair(mut self, keypair: Keypair) -> Self {
```

Comment:
Please add a doc comment.

Also looking at `NetworkBuilder::build`, please add that a keypair will be randomly generated if not set.